functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -293,6 +293,42 @@ oneOutput matrix_exp_batch_rule(const Tensor& self, c10::optional<int64_t> self_
   return std::make_tuple(at::matrix_exp(self_), 0);
 }
 
+ fourOutputs solve_ex_batch_rule(
+    const Tensor& A, optional<int64_t> A_bdim,
+    const Tensor& B, optional<int64_t> B_bdim,
+    bool left, bool check_errors) {
+  auto batch_size = get_bdim_size2(A, A_bdim, B, B_bdim);
+  const auto A_logical_rank = rankWithoutBatchDim(A, A_bdim);
+  const auto B_logical_rank = rankWithoutBatchDim(B, B_bdim);
+  const auto max_logical_rank = std::max(A_logical_rank, B_logical_rank);
+
+  TORCH_CHECK(A_logical_rank >= 2,
+            "linalg.solve: The input tensor A must have at least 2 dimensions.");
+
+  int b_logical_rank = max_logical_rank;
+  if (A_logical_rank > B_logical_rank) {  // vector case: B was a vector or batched vector
+    // not accurate but matches linalg error message
+    TORCH_CHECK(B_logical_rank >= 1, "linalg.solve: The input tensor B must have at least 2 dimensions.");
+    b_logical_rank = max_logical_rank - 1;
+  } else {  // matrix case: A and B are both matrices or batches of matrices
+    TORCH_CHECK(B_logical_rank >= 2, "linalg.solve: The input tensor B must have at least 2 dimensions.");
+  }
+
+  // basically binary pointwise helper but if B was a vector incoming, we must pad it to be 1 dim smaller than A
+  auto A_ = moveBatchDimToFront(A, A_bdim);
+  auto B_ = moveBatchDimToFront(B, B_bdim);
+  A_ = maybePadToLogicalRank(A_, A_bdim, max_logical_rank);
+  B_ = maybePadToLogicalRank(B_, B_bdim, b_logical_rank);
+
+  A_ = ensure_has_bdim(A_, A_bdim.has_value(), batch_size);
+  B_ = ensure_has_bdim(B_, B_bdim.has_value(), batch_size);
+  if (!A_bdim.has_value()) {
+    A_ = A_.contiguous();  // bug?
```

Comment:
Do you have a repro that doesn't involve vmap? This would imply that solve_ex has problems with non-contiguous inputs and we should file an issue if so

Reply:
I haven't been able to but I'll keep trying to and looking at this more...I'm now wondering if it's something around assuming that the optimization was used during the forward when doing autograd since it only came up when we would have had `GradTensor(BatchedTensor(...))`. If this is right, I can break down my theory more since it's kind of confusing right now